go-controller/pkg/ovn/controller/apbroute/external_controller.go

```diff
@@ -128,45 +153,92 @@ func newExternalPolicyManager(
 	netClient networkClient) *externalPolicyManager {
 
 	m := externalPolicyManager{
-		stopCh:                 stopCh,
-		routeLister:            routeLister,
-		podLister:              podLister,
-		namespaceLister:        namespaceLister,
-		namespaceInfoSyncCache: syncmap.NewSyncMap[*namespaceInfo](),
-		routePolicySyncCache:   syncmap.NewSyncMap[*adminpolicybasedrouteapi.AdminPolicyBasedExternalRoute](),
-		netClient:              netClient,
+		stopCh:                         stopCh,
+		routeLister:                    routeLister,
+		podLister:                      podLister,
+		namespaceLister:                namespaceLister,
+		namespaceInfoSyncCache:         syncmap.NewSyncMap[*namespaceInfo](),
+		routePolicySyncCache:           syncmap.NewSyncMap[*routeInfo](),
+		netClient:                      netClient,
+		mutexRoutePolicyCachePopulated: &sync.Mutex{},
 	}
 
 	return &m
 }
 
+func (m *externalPolicyManager) setRoutePolicyCacheAsPopulated() {
+	m.mutexRoutePolicyCachePopulated.Lock()
+	defer m.mutexRoutePolicyCachePopulated.Unlock()
+	m.routePolicyCachePopulated = true
+}
+
+func (m *externalPolicyManager) isRoutePolicyCachePopulated() bool {
+	m.mutexRoutePolicyCachePopulated.Lock()
+	defer m.mutexRoutePolicyCachePopulated.Unlock()
+	return m.routePolicyCachePopulated
+}
+
 // getRoutePolicyFromCache retrieves the cached value of the policy if it exists in the cache, as well as locking the key in case it exists.
-func (m *externalPolicyManager) getRoutePolicyFromCache(policyName string) (*adminpolicybasedrouteapi.AdminPolicyBasedExternalRoute, bool) {
+func (m *externalPolicyManager) getRoutePolicyFromCache(policyName string) (*adminpolicybasedrouteapi.AdminPolicyBasedExternalRoute, bool, bool) {
```

Comment:
do you really need to return "policy" and "found". If policy is nil, doesnt that mean it isnt found? Or can a policy actually be stored as nil?